.devcontainer/devcontainer.json

```diff
@@ -11,6 +11,11 @@
 		"tamasfe.even-better-toml",
 		"zxh404.vscode-proto3"
 	],
+	"settings": {
+		"rust-analyzer.server.extraEnv": {
+			"RUSTFLAGS": "-D warnings -A deprecated"
```

Comment:
i'm...not totally sure how i feel about setting `-D warnings` in dev builds? i know the `justfile` already does it (https://github.com/linkerd/linkerd2-proxy/blob/f3cf2e8b8556502bfbf7138d9b26517ff0fd6afe/justfile#L8) but i'm not sure if we should be doing that; i would like to be able to run tests from the justfile even if the proxy builds with warnings, while working on local dev branches. maybe we should change this to only add `-D warnings` if on CI? idk.

in any case, that should probably be addressed separately; if we *are* going to set `-D warnings` in `just test`, we should do it in rust-analyzer as well.